src/proc_macros/gat-attr/_mod.rs

```diff
@@ -2,17 +2,27 @@ use super::*;
 
 mod trait_def;
 mod trait_impl;
+mod trait_use;
 
 pub(in super)
 fn gat (
     attrs: TokenStream2,
     input: TokenStream2,
 ) -> Result<TokenStream2>
 {
-    let _: parse::Nothing = parse2(attrs)?;
     match parse2(input)? {
-        | Item::Trait(item_trait) => trait_def::handle(item_trait),
-        | Item::Impl(item_impl) => trait_impl::handle(item_impl),
+        | Item::Trait(item_trait) => {
+            let _: parse::Nothing = parse2(attrs)?;
+            trait_def::handle(item_trait)
+        },
+        | Item::Impl(item_impl) => {
+            let _: parse::Nothing = parse2(attrs)?;
+            trait_impl::handle(item_impl)
+        },
+        | Item::Use(item_use) => {
+            let assoc_types = Punctuated::<NestedMeta, syn::token::Comma>::parse_terminated.parse2(attrs)?;
```

Comment:
```suggestion
            let assoc_types = Punctuated::<Ident, Token![,]>::parse_terminated.parse2(attrs)?;
```

Using `Ident` will allow us to get rid of the extra bails when unwrapping the expected single `Ident` out of the `NestedMeta` (and the `Token![…]` is just a style preference 🙃)

Reply:
> Using Ident will allow us to get rid of the extra bails

haha, it does make the code a lot nicer; had a bit of tunnel vision "oh I'm parsing a meta item, so I should use a `*Meta` type".

> `Token![…]` is just a style preference

happy to adapt :v: